src/Microsoft.OData.Edm/Schema/Interfaces/IEdmFullNamedElement.cs

```diff
@@ -0,0 +1,19 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="IEdmFullNamedElement.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// Common base interface for all named EDM elements.
+    /// </summary>
+    public interface IEdmFullNamedElement : IEdmNamedElement

```

Comment:
Any reason why base type of IEdmNamedElement is required?

Reply:
Only Named Elements should have FullName.

Reply:
Are there cases where a named element wouldn't have a full name (qualified name?) associated as well? I'm wondering if the new FullName property would logically belong in IEdmNamedElement.

Might need input from @xuzhg / @mikepizzo on this.

Reply:
Cant change public interface as it will be breaking change.

Reply:
I try to think why we want to add a new interface like this? 

ODL.Edm lib has a full defined structured for element. And I don't think we should have this new interface to cache the full name.





---
In reply to: [209776151](https://github.com/OData/odata.net/pull/1227#discussion_r209776151) [](ancestors = 209776151)

Reply:
Is there a proposed alternative?

Reply:
I am still reading the PR. Can we use the property bag in the model to cache the mapping between element and full name?

---
In reply to: [211420888](https://github.com/OData/odata.net/pull/1227#discussion_r211420888) [](ancestors = 211420888)

Reply:
We can but then it doesn't put us in the path forward. Ideally when ODL does it's next breaking change release, the interface I added should be removed and this property should be moved to one of the existing interfaces (read INamedElement). This way it belong to where it needs to belong. No need add additional property bag and stuff. My change puts towards this path forward.

Reply:
As Yogesh says, ideally we would put fullname in IEdmNamedElement and just use that, but since we can't make breaking changes it makes sense to have it in a derived interface.

---
In reply to: [211419623](https://github.com/OData/odata.net/pull/1227#discussion_r211419623) [](ancestors = 211419623,209776151)

Reply:
Agree. Keeping the full name on the type instance is much simpler/more straight forward than creating a side mechanism.

---
In reply to: [211429169](https://github.com/OData/odata.net/pull/1227#discussion_r211429169) [](ancestors = 211429169)